src/utilities/utilities.jl

```diff
@@ -139,3 +139,26 @@ end
 function _assignment_to_kw!(x)
     return throw(ArgumentError("Expected assignment expression, instead found: `$x`"))
 end
+
+# https://docs.aws.amazon.com/sdkref/latest/guide/feature-retry-behavior.html
+struct AWSExponentialBackoff
+    max_attempts::Int
+    max_backoff::Float64
+end
+
+# Defaults for AWS's `standard` retry mode. Note: these can be overridden elsewhere.
+function AWSExponentialBackoff(; max_attempts=3, max_backoff=20.0)
+    return AWSExponentialBackoff(max_attempts, max_backoff)
+end
+
+# We make one more attempt than the number of delays
+Base.length(exp::AWSExponentialBackoff) = exp.max_attempts - 1
+
+function Base.iterate(exp::AWSExponentialBackoff, i=1)
+    i >= exp.max_attempts && return nothing
+    # rand() has values in [0, 1), so we use 1.0 - rand() which has values in (0, 1] required.
+    b = 1.0 - rand()
```

Comment:
Should pass in an `rng` into this for testing purposes
```suggestion
    b = 1.0 - rand(exp.rng)
```